lib/mixlib/install/backend/artifactory.rb

```diff
@@ -19,72 +19,141 @@
+        #
+        def info
+          if options.platform
+            product = products.find do |data|
+              data["omnibus.platform"] == options.platform &&
+                data["omnibus.platform_version"] == options.platform_version &&
+                data["omnibus.architecture"] == options.architecture
+            end
+            # TODO: Handle no find results
+            artifact(product)
+          else
+            products.collect do |data|
+              artifact(data)
+            end
+          end
+        end
 
-        ARTIFACTORY_REPOSITORY = "omnibus-unstable-local".freeze
+        #
+        # Return Array<Hash> containing all queried artifacts for channel, product_name,
+        #   and product_version.
+        # The Hash will contain:
+        #    download_uri: The full url download path
+        #    <property_name>: The names of the properties associcated to the artifact
+        #
+        def products
+          query = <<-QUERY
+items.find(
+  {"repo": "#{repository}"},
+  {"@omnibus.project": "#{options.product_name}"},
+  {"@omnibus.version": "#{product_version}"}
+).include("name", "path", "repo", "property")
+          QUERY
 
-        attr_reader :options
-        attr_reader :client
+          results = aql_post(query)
 
-        def initialize(options)
-          @options = options
-          @client = ::Artifactory::Client.new(endpoint: ARTIFACTORY_ENDPOINT)
+          results["results"].collect do |result|
+            uri = [::Artifactory.endpoint]
+            uri << result["repo"]
+            uri << result["path"]
+            uri << result["name"]
+            { "downloadUri" => uri.join("/") }.merge(map_properties(result["properties"]))
+          end
         end
 
-        def info
-          begin
-            results = client.get("/api/search/prop", params, headers)["results"]
-          rescue Errno::ETIMEDOUT => e
-            raise e, "unstable channel uses endpoint #{ARTIFACTORY_ENDPOINT} \
-which is currently only accessible through Chef's internal network."
+        #
+        # Returns all version found for channel and product_name
+        #
+        def product_versions
+          query = <<-QUERY
+items.find(
+  {"repo": "#{repository}"},
+  {"@omnibus.project": "#{options.product_name}"}
+).include("name", "path", "repo", "property")
+          QUERY
+
+          results = aql_post(query)
+          properties = results["results"].collect { |result| result["properties"] }
+          versions = properties.collect do |prop|
+            prop.first["value"] if prop.first["key"] == "omnibus.version"
           end
+          versions.compact
+        end
 
-          if options.platform
-            artifact(results.first)
+        #
+        # Parse latest version
+        #
+        def product_version
+          if options.latest_version?
+            Mixlib::Versioning.find_target_version(product_versions, nil, false, true).to_s
```

Comment:
We should have some documentation here about the parameters we are passing in to `find_target_version`.